backend/src/main/java/ca/bc/gov/hlth/mohums/webclient/KeycloakApiService.java

```diff
@@ -0,0 +1,134 @@
+package ca.bc.gov.hlth.mohums.webclient;
+
+import ca.bc.gov.hlth.mohums.model.Group;
+import ca.bc.gov.hlth.mohums.model.GroupDescriptionGenerator;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.MultiValueMap;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class KeycloakApiService {
+
+    private final String clientsPath = "/clients";
+    private final String usersPath = "/users";
+    private final String groupsPath = "/groups";
+    private final String userClientRoleMappingPath = "/role-mappings/clients/";
+
+    private final ExternalApiCaller externalApiCaller;
+
+    public KeycloakApiService(@Qualifier("keycloakApiCaller") ExternalApiCaller externalApiCaller) {
+        this.externalApiCaller = externalApiCaller;
+    }
+
+    // Clients
+    public ResponseEntity<List<Object>> getClients() {
+        return externalApiCaller.getList(clientsPath);
+
+    }
+
+    public ResponseEntity<Object> getClient(String clientId) {
+        String path = clientsPath + "/" + clientId;
+        return externalApiCaller.get(path, null);
+    }
+
+    public ResponseEntity<List<Object>> getClientRoles(String clientId) {
+        String path = String.format("%s/%s/roles", clientsPath, clientId);
+        return externalApiCaller.getList(path, null);
+    }
+
+    public ResponseEntity<List<Object>> getUsersInRole(String clientId, String roleName,
+                                                       MultiValueMap<String, String> queryParams) {
+        String path = String.format("%s/%s/roles/%s/users", clientsPath, clientId, roleName);
+        return externalApiCaller.getList(path, queryParams);
+    }
+
+    // Groups
+    public ResponseEntity<Object> getGroups() {
+        ResponseEntity<Object> response = externalApiCaller.get(groupsPath, null);
+        ArrayList<LinkedHashMap> groups = (ArrayList<LinkedHashMap>) response.getBody();
```

Comment:
Would it be worth having some error handling here incase that get happens to fail?